Tests/scripts/collect_tests/collect_tests.py

```diff
@@ -438,14 +471,22 @@ def _get_changed_files(self) -> tuple[str, ...]:
             logger.info(f'contribution branch, {contrib_diff=}')
 
         diff: str = run_command(f'git diff --name-status {current_commit}...{previous_commit}')
-        logger.debug(f'Changed files: {diff}')
+        logger.debug(f'Changed files:\n{diff}')
 
         if contrib_diff:
             logger.debug('adding contrib_diff to diff')
             diff = f'{diff}\n{contrib_diff}'
 
         # diff is formatted as `M  foo.json\n A  bar.py\n ...`, turning it into ('foo.json', 'bar.py', ...).
-        return tuple((value.split()[1] for value in filter(None, diff.split('\n'))))
+        files = []
+        for line in filter(None, diff.splitlines()):
+            status, path = line.split()
+            if status != 'D':
```

Comment:
status D?

Reply:
deleted, explained in the following rows, i'll move the comment here for clarity.